server/src/test/java/io/druid/segment/realtime/plumber/RealtimePlumberSchoolTest.java

```diff
@@ -297,4 +319,144 @@ public void run()
 
     Assert.assertEquals(1, metrics.failedPersists());
   }
+
+  @Test(timeout = 60000)
+  public void testPersistHydrantGaps() throws Exception
+  {
+    final Object commitMetadata = "dummyCommitMetadata";
+    testPersistHydrantGapsHelper(commitMetadata);
+  }
+
+  private void testPersistHydrantGapsHelper(final Object commitMetadata) throws Exception
+  {
+    final MutableBoolean committed = new MutableBoolean(false);
```

Comment:
Shouldn't this be an AtomicBoolean? The run() of the committer happens in a different thread.